### ISSUE TITLE
New Feature: Show Toolbars Only When The Mouse Hovered In

### ISSUE THREAD

**Issue by mistgc** [gabm/Satty#42]
The toolbars don't display if the mouse doesn't hover in the app window.
<img src="https://github.com/gabm/Satty/assets/61856080/b9d6f49b-abf4-4307-abb6-b88014c056e6" width="300" height="200"/>
And show the toolbars when we hover in.
<img src="https://github.com/gabm/Satty/assets/61856080/15ddaded-0df0-4980-a904-c785b6ce4fe7" width="300" height="200"/>

**Comment by gabm** [comment]
thank you for that idea.. could you explain what we try to achieve with this feature?

**Comment by mistgc** [comment]
Imagine a scenario where a programmer is writing code while looking at other people's code snippets.
Like this:
![image](https://github.com/gabm/Satty/assets/61856080/7be3f16e-be90-4afc-863b-57d7997d1136)

**Comment by mistgc** [comment]
We could pin some screenshots on the screen.

**Comment by mistgc** [comment]
While this is true, we can add annotations to the image.

**Comment by gabm** [comment]
okay.. well I think there is nothing wrong with hiding the toolbars on demand. I would prefer a (non-default) shortcut for this, like `Ctrl-T` for "toggle toolbar" or something like this.

I must say that I am a big fan of explicit behaviour (shortcut/button toggles toolbar) and not dynamic behaviour (mouse hover toggles toolbar). Further I think that should be "opt-in" .. not the default behaviour. There might be a setting "hide-toolbar-on-startup" or something similar..

I am happy to accept PRs

**Comment by mistgc** [comment]
I implemented the basic functionality, but there isn't support for configuration.

**Comment by mistgc** [comment]
Now, it supports configuration.
BUT, it doesn't support hovering to show toolbars.

**Comment by KAGEYAM4** [comment]
What are your thoughts on Krita like toolbar which gets opened using right-mouse-button ?
![image](https://github.com/gabm/Satty/assets/75798544/9c6f7a3a-1fb3-4ae0-81eb-2096354de010)

**Comment by mistgc** [comment]
I think most people find this very convenient, but for users who frequently use keyboard shortcuts (such as Window Manager, Vim-like Editor, Emacs), it is not convenient.

**Comment by KAGEYAM4** [comment]
You're right, Plus it's will increase development time which would be wasteful as this dosen't offer much as keyboard based implementation will still be there plus the hovering toolbar.

**Comment by alteriks** [comment]
I like the way that flameshot took. 'toolbar' is outward of the image.
![image](https://github.com/gabm/Satty/assets/6512767/e06c9227-0a33-4fb7-a3e2-2d6398f2b372)
Buttons reflow if the selection was made near the screen edge.
![image](https://github.com/gabm/Satty/assets/6512767/59fd5bc1-39ff-4ba4-ac86-aef1699fc502)

**Comment by gabm** [comment]
how does it work when you want to edit the fullscreen image, e.g. without crop?

**Comment by alteriks** [comment]
It's been ages since I took fullscreen shot. Using `flameshot screen`  asks only about saving file. I don't recall if you could edit fullscreen shot after it had been taken.

**Comment by gabm** [comment]
the first basic functionality has been released as https://github.com/gabm/Satty/releases/tag/v0.10.0

**Comment by RobertMueller2** [comment]
As we now have both a hotkey AND mouse over to toggle, I'm closing this issue.
